services/explorer/src/pages/Faucet/Faucet.effects.ts

```diff
@@ -0,0 +1,105 @@
+import { useSnackbar } from 'notistack';
+import { useCallback, useEffect, useState } from 'react';
+import { faucetUri } from '../../config';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+
+export enum RequestLimit {
+  ONE = 1,
+  TWO = 2,
+  FIVE = 5,
+  SEVEN = 7,
+}
+export const useFaucetEffects = () => {
+  const { currentNetwork } = useExplorerNetwork();
+  const { isLoading, initLoading, finishLoading } = useExplorerLoader();
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const [address, setAddress] = useState<string>('');
+  const [amount, setAmount] = useState<RequestLimit>(1);
+  const [displayToast, setDisplayToast] = useState<{
+    display: boolean;
+    variant?: 'success' | 'error' | 'warning' | undefined;
+    text?: string;
+  }>({ display: false, variant: undefined, text: '' });
+
+  const sendTokenRequest = useCallback(() => {
+    const requestTokens = async () => {
+      initLoading();
+      try {
+        const response = await fetch(faucetUri, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            address,
+            amount,
+            timestamp: Date.now() / 1000,
```

Comment:
We need to ensure that the timezone sent from the frontend is the same as the one in the backend. If the user is in GMT+2 but the server is in the UTC, we need to know this when comparing the dates. This can be done directly on the backend, but I thought it is worth mentioning it here so we don't forget.
Anyways, by doing Date.now() and dividing it by 1000 you might loose the timezone. Maybe we can send directly the date (new Date()) and handle the conversion on the backend.